### PR TITLE
[GEOS-9461] Update YAML docs for new cascaded WMS controls

### PR DESCRIPTION
**Pull request by imranrajjad**
JIRA : https://osgeo-org.atlassian.net/browse/GEOS-9461

update \doc\en\api\1.0.0\wmtslayers.yaml

-add remote styles and format
-add min/max scale denominators
-add respect remote bounds settings

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [x] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [x] Committs changing the REST API, or any configuration object, should check it the REST API docs (Swagger YAML files and classic documentation) need to be updated.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: doc/en/api/1.0.0/wmslayers.yaml [hunk 1]
```diff
@@ -139,55 +139,109 @@ paths:
...
-                <projectionPolicy>FORCE_DECLARED</projectionPolicy>
-                <enabled>true</enabled>
-                <metadata>
-                  <entry key="cachingEnabled">false</entry>
-                </metadata>
-                <store class="wmsStore">
-                  <name>cite:altgs</name>
-                  <atom:link xmlns:atom="http://www.w3.org/2005/Atom" rel="alternate" href="http://localhost:8080/geoserver/rest/workspaces/cite/wmsstores/altgs.xml" type="application/xml"/>
-                </store>
+                 <name>dem</name>
```
Comment (jodygarnett) [inline doc/en/api/1.0.0/wmslayers.yaml]: Some small difference of indenting here, just making a note that it masks the small change you performed in the example.

File: doc/en/api/1.0.0/wmslayers.yaml [hunk 2]
```diff
@@ -139,55 +139,109 @@ paths:
...
+                          <version>1.0.0</version>
+                       </languageVersion>
+                       <legend>
+                          <width>20</width>
+                          <height>20</height>
+                          <format>image/png</format>
+                          <onlineResource>https://remote.wms.server/geoserver/wms?request=GetLegendGraphic&amp;format=image%2Fpng&amp;width=20&amp;height=20&amp;layer=topp%3Astates&amp;style=dem_style</onlineResource>
+                       </legend>
+                       <metadata>
+                          <entry key="isRemote">true</entry>
```
Comment (jodygarnett) [inline doc/en/api/1.0.0/wmslayers.yaml]: excellent!

File: doc/en/api/1.0.0/wmslayers.yaml [hunk 3]
```diff
@@ -746,4 +830,41 @@ definitions:
...
+            version:
+              type: string
+              description: versioon of format
+              example: "1.0.0"
+        legend:
+          type: object
+          properties:
+            width:
+              type: number
+              description: version of format
```
Comment (jodygarnett) [inline doc/en/api/1.0.0/wmslayers.yaml]: description being repeated here

File: doc/en/api/1.0.0/wmslayers.yaml (hunk 1 above)
Comment (imranrajjad) [reply doc/en/api/1.0.0/wmslayers.yaml]: will fix

File: doc/en/api/1.0.0/wmslayers.yaml (hunk 3 above)
Comment (imranrajjad) [reply doc/en/api/1.0.0/wmslayers.yaml]: will fix

**Comment by imranrajjad** [comment]
I am unable to identify the cause of documentation build failure here

```
[INFO] reading from /home/travis/build/geoserver/geoserver/doc/en/api/1.0.0/wmslayers.yaml
387
[ERROR] failed to read resource listing
388
com.fasterxml.jackson.core.JsonParseException: Unexpected character ('-' (code 45)) in numeric value: expected digit (0-9) to follow minus sign, for valid numeric value
```

@jodygarnett can you please help me here, not a YAML expert myself

**Comment by jodygarnett** [comment]
> I am unable to identify the cause of documentation build failure here
> 
> ```
> [INFO] reading from /home/travis/build/geoserver/geoserver/doc/en/api/1.0.0/wmslayers.yaml
> 387
> [ERROR] failed to read resource listing
> 388
> com.fasterxml.jackson.core.JsonParseException: Unexpected character ('-' (code 45)) in numeric value: expected digit (0-9) to follow minus sign, for valid numeric value
> ```
> 
> @jodygarnett can you please help me here, not a YAML expert myself

I use https://editor.swagger.io and work there. @groldan is also very helpful :)

**Comment by imranrajjad** [comment]
@jodygarnett I have verified the file at https://editor.swagger.io/ , it shows as valid

**Comment by jodygarnett** [comment]
So the error is:
```
[ERROR] failed to read resource listing
390com.fasterxml.jackson.core.JsonParseException: Unexpected character ('-' (code 45)) in numeric value: expected digit (0-9) to follow minus sign, for valid numeric value
391 at [Source: /home/travis/build/geoserver/geoserver/doc/en/api/1.0.0/wmslayers.yaml; line: 1, column: 3]
```

That location has:

```
---
swagger: '2.0'
info:
  version: 1.0.0
  title: GeoServer WMS Store Layers
  description: A WMS store is a store whose source is another WMS. Also known as "Cascading WMS" or "External WMS". A WMS store layer is a layer from this store.
```

I wonder if you only need two `--`

**Comment by jodygarnett** [comment]
Okay trying it on a web YAML validator: http://www.yamllint.com

Shows the following:

```
            format:
              type: string
              description: image format of legend graphic
              example: "image\/png"
            onlineResource:
              type: string
              description: GetLegendGraphic request URL
```

> (<unknown>): found unknown escape character while parsing a quoted scalar at line 864 column 24

So `\\` is used to escape, and the escape character ``\\/`` does not mean anything?

**Comment by imranrajjad** [comment]
@jodygarnett well that did the trick, although the JSON error is completely misleading

good to merge?

**Comment by jodygarnett** [comment]
+1 from me:
- can you go through the checklist
- we need to squash the commits (we can do that using "squash and merge")

**Comment by imranrajjad** [comment]
@jodygarnett done

**Comment by aaime** [comment]
@imranrajjad merging, will you handle the backport?

**Comment by imranrajjad** [comment]
yes I will

**Pull request merged**